File: Python/023.py
from Utils import sum_of_proper_divisors
import time

start = time.time()
abundant = [i for i in range(2,28123) if sum_of_proper_divisors(i)-i>i]

sums = {i:False for i in range(28123)}

for i in abundant:
    print(i)

for i in range(len(abundant)):
    for j in range(i,len(abundant)):
        sum_abundant = abundant[i]+abundant[j]
        sums[sum_abundant] = True
        if sum_abundant > 28123:
            break

print(sum([k for k,v in sums.items() if not v]))
print(time.time()-start)

File: C++/023.cpp
#include <iostream>
#include <vector>
#include "Utils.h"

using namespace std;

int main() {
  
  int limit = 28123;
  vector<int> abundant;
  bool sums[limit+1];
  for (int i=12; i<limit; i++) {
    int s = sumProperDivisors(i)-i;
    if (s > i)
      abundant.push_back(i);
  }

  for (int i=0; i<abundant.size(); i++) {
    for (int j=i; j<abundant.size(); j++) {
      int sumsAbundant = abundant[i] + abundant[j];
      if (sumsAbundant > limit)
        break;
      sums[sumsAbundant] = true;
    }
  }

  int res = 0;
  for (int i=1; i<=limit; i++) {
    if (sums[i] == false)
      res += i;
  }
  cout << res << "\n";
}
